processor/otel/consumer.go

```diff
@@ -611,3 +618,21 @@ func truncate(s string) string {
 	}
 	return s
 }
+
+type transformableError struct {
+	*apmerror.Event
+}
+
+func (te transformableError) Transform() []beat.Event {
+	// TODO(axw) provide different methods on Error for transforming RUM vs. non-RUM errors?
+	return te.Event.Transform(nil, nil, nil)
+}
+
+type transformableSpan struct {
+	*span.Event
+}
+
+func (ts transformableSpan) Transform() []beat.Event {
+	// TODO(axw) provide different methods on Span for transforming RUM vs. non-RUM errors?
+	return ts.Event.Transform(nil, nil, nil)
+}
```

Comment:
Why duplicating the logic here instead of making the `transformableError` and `transformableSpan` at https://github.com/elastic/apm-server/pull/3254/files#diff-9544c869a4bceb063ad78365bfa27c88 public?

Reply:
that makes sense, thanks for the explanation.